### PR TITLE
[CDS-36615]: serverless prepare rollback tech debt

### PR DESCRIPTION
**Pull request by PragyeshMishra01**
## Describe your changes
-- Added a new serverless task for preparing rollback data to handle corner-case rollback scenerios.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34180)

### CODE REVIEW

File: 950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java [hunk 1]
```diff
@@ -351,7 +351,7 @@ public enum TaskType {
   CLOUDFORMATION_TASK_NG(TaskGroup.CLOUDFORMATION_NG, "Cloudformation Task"),
   ACR_ARTIFACT_TASK_NG(TaskGroup.ARTIFACT_COLLECT_NG, "ACR Task"),
   SERVERLESS_GIT_FETCH_TASK_NG(TaskGroup.SERVERLESS_NG, "Git Fetch Files Task"),
```
Comment (sainath-b) [inline 950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java:353]: Name it serverless git fetch files task

Comment (PragyeshMishra01) [reply 950-delegate-tasks-beans/src/main/java/software/wings/beans/TaskType.java]: done

**Comment by PragyeshMishra01** [comment]
trigger messagecheck

**Comment by PragyeshMishra01** [comment]
trigger messagecheck

**Review by sainath-b — APPROVED** [review]

**Comment by PragyeshMishra01** [comment]
trigger ft2

**Comment by PragyeshMishra01** [comment]
trigger compile

**Pull request merged**